custom/dhis2/models.py

```diff
@@ -179,63 +242,58 @@ def add_te_inst(self, entity_data, te_name, ou_id):
 
 
         """
-        # Convert data keys to tracked entity attribute IDs
-        if any(key not in self._tracked_entity_attributes for key in entity_data):
-            # We are going to have to fetch at least one tracked entity attribute ID. Fetch them all to avoid
-            # multiple API requests.
-            self._fetch_tracked_entity_attributes()
-        # Create a list of attributes, looking up the attribute ID of each one
-        attributes = [{
-            'attribute': self.get_te_attr_id(key),
-            'value': value
-        } for key, value in entity_data.iteritems()]
         request_data = {
             'trackedEntity': self.get_te_id(te_name),
             'orgUnit': ou_id,
-            'attributes': attributes
+            'attributes': self._data_to_attributes(instance_data)
         }
         __, response = self._request.post('trackedEntityInstances', request_data)
```

Comment:
Doesn't seem like you ever check the status code returned by `JsonApiRequest`. Maybe consider just not returning it?